exp/internal/controllers/machinepool_controller.go

```diff
@@ -41,6 +41,7 @@ import (
 	"sigs.k8s.io/cluster-api/controllers/external"
 	"sigs.k8s.io/cluster-api/controllers/remote"
 	expv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/cluster-api/internal/util/ssa"
```

Comment:
are these changes overlapping with #9619 ? If so can we rebase on top of it so it's easier for review to only look at  the changes this PR is adding

I think it would also help to cleanup commit history for review so the reviewers can follow logical commits. If you don't want to lose the history before the refactor, you can squash everything "pre refactor" into one and then keep only the commits you made recently to change things so it can be easier to see what changed.

Reply:
Done, I think that should make the Docker PR changes a bit smaller. Thanks for the suggestion!